### ISSUE TITLE
flatpak history returns error

### ISSUE THREAD

**Issue by agriffis** [flatpak/flatpak#4121]
## Linux distribution and version

Fedora Silverblue 34.20210211.n.1

## Flatpak version

1.10.1

## Description of the problem

```
$ flatpak history
error:  is not application or runtime
```

## Steps to reproduce

1. Run `flatpak history`

**Comment by agriffis** [comment]
Here's another flavor

```
$ flatpak history --reverse
error: appstream2/x86_64 is not application or runtime
```

**Comment by luni3359** [comment]
> Here's another flavor
> 
> ```
> $ flatpak history --reverse
> error: appstream2/x86_64 is not application or runtime
> ```

I get this error both with `flatpak history` and `flatpak history --reverse`.

**Comment by harinath** [comment]
Appears to be still present in version 1.11.1

---8<---
❯ flatpak history 
error: appstream2/x86_64 is not application or runtime

~ on ☁️  harinath@gmail.com(asia-south1) 
❯ flatpak --version
Flatpak 1.11.1
---8<---

**Comment by luni3359** [comment]
Can confirm, it's still an issue. 

KDE neon User Edition 5.21 x86_64 
Flatpak 1.11.1

```
$ flatpak history 
error: appstream2/x86_64 is not application or runtime
```

**Comment by hackel** [comment]
Still present on 1.11.2.

**Comment by repsorp** [comment]
6 months old now...
any other way to access history ?

```
$ flatpak history 
error: appstream2/x86_64 is not application or runtime
```


config:
Ubuntu 21.04

**Comment by repsorp** [comment]
https://github.com/flatpak/flatpak/pull/4180#issuecomment-913124721

**Comment by LinuxOpa** [comment]
> 6 months old now... any other way to access history ?
> 
> ```
> $ flatpak history 
> error: appstream2/x86_64 is not application or runtime
> ```
> 
> config: Ubuntu 21.04

me too
$ flatpak history 
error: appstream2/x86_64 is not application or runtime

Ubuntu 20.04.3 LTS

**Comment by tim77** [comment]
Still on Fedora 35. `flatpak-1.11.3-1`.

**Comment by DMRobertson** [comment]
I'm seeing this on Fedora 35 with Flatpak 1.12.2. Happy to provide any extra information that's useful!

```
 $ flatpak --version; flatpak history
Flatpak 1.12.2
error: appstream2/x86_64 is not application or runtime
```

**Comment by mwleeds** [comment]
@DMRobertson the fix is in #4526 waiting on code review

**Comment by DMRobertson** [comment]
Ahh, apologies; I hadn't seen that or #4332. Many thanks!

> @DMRobertson the fix is in #4526 waiting on code review

**Comment by Zorono** [comment]
still happening on Flatpak 1.10.2 (Ubuntu 21.10)

$ **flatpak history**:
```
error: appstream2/x86_64 is not application or runtime
```

$ **flatpak history -v**:
```
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/{USER}/.local/share/flatpak
error: appstream2/x86_64 is not application or runtime
```

**Comment by mwleeds** [comment]
The 1.10.x backport is in #4729

**Comment by smcv** [comment]
> still happening on Flatpak 1.10.2 (Ubuntu 21.10)

Ubuntu 21.10 users will probably not get this change until they upgrade to Ubuntu 22.04, even if we do a new 1.10.x release with the fixes backported, because Ubuntu don't seem to be willing to pick up our bugfix releases like 1.10.3.

**Comment by Zorono** [comment]
> > still happening on Flatpak 1.10.2 (Ubuntu 21.10)
> 
> Ubuntu 21.10 users will probably not get this change until they upgrade to Ubuntu 22.04, even if we do a new 1.10.x release with the fixes backported, because Ubuntu don't seem to be willing to pick up our bugfix releases like 1.10.3.

jammy isn't going to get released before **April 21** :(
really i hate how Debian-based repositories are working!!!!

**Comment by smcv** [comment]
> really i hate how Debian-based repositories are working

Don't blame Debian for this: 1.10.x updates do get into Debian point releases (because I do the necessary work).

**Comment by cdplayer** [comment]
Still occurring with Flatpak 1.12.2
```
$ flatpak history -v
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /root/.local/share/flatpak
error: appstream2/x86_64 is not application or runtime
```

**Comment by smcv** [comment]
> Still occurring with Flatpak 1.12.2

This is believed to be fixed in [1.12.5](https://github.com/flatpak/flatpak/releases/tag/1.12.5) (the current 1.12.x release is 1.12.7). If your distribution doesn't take updates from the 1.12.x bugfix-only branch, please ask them to do that.
